pages/my-orders.js

```diff
@@ -4,17 +4,46 @@ import Layout from '../components/layout'
 import Navbar from '../components/navbar'
 import Table from '../components/table'
 import { getOrders } from '../data/orders'
+import { getSinglePaymentType } from '../data/payment-types'
 
 export default function Orders() {
   const [orders, setOrders] = useState([])
   const headers = ['Order Date', 'Total', 'Payment Method']
 
+  const fetchPaymentTypes = async (url) => {
+      const response = await getSinglePaymentType(url);
+      if (response === "No payment made") {
+        return response;
+      }
+      const paymentData = await response.json();
+      return paymentData.merchant_name;
+  }
+
+  const caculateTotal = (productArray) => {
+    const totalAmount = 0
+    productArray.map(item => (totalAmount += item.product.price))
+    return totalAmount
+  }
+
   useEffect(() => {
```

Comment:
The current use of useEffect to define and immediately call a large async function does not seem ideal. It makes the component harder to read and test.